runtime/onert/backend/trix/DevContext.cc

```diff
@@ -225,30 +238,57 @@ void DevContext::runOneBatch(uint32_t dev_num, ModelID model_id, input_buffers *
 
   const auto &dev_handle = _dev_handles.at(dev_num);
   int req_id;
-  if (createNPU_request(dev_handle, model_id_at_device, &req_id))
+
+  if (auto error_code = createNPU_request(dev_handle, model_id_at_device, &req_id))
   {
     throw std::runtime_error("Unable to create NPU request with model id (" +
-                             std::to_string(model_id_at_device) + ")");
+                             std::to_string(model_id_at_device) + ")" +
+                             " error code : " + std::to_string(error_code));
   }
 
-  if (setNPU_requestData(dev_handle, req_id, input_bufs, in_info, output_bufs, out_info))
+  if (auto error_code =
+        setNPU_requestData(dev_handle, req_id, input_bufs, in_info, output_bufs, out_info))
   {
     removeNPU_request(dev_handle, req_id);
     throw std::runtime_error("Unable to create NPU request for model id (" +
-                             std::to_string(model_id_at_device) + ")");
+                             std::to_string(model_id_at_device) + ")" +
+                             " error code : " + std::to_string(error_code));
+  }
+
+  // NOTE submitNPU_request is not thread-safe(?). It is rarely hanging(unresponsive).
```

Comment:
(optional) need to ask to trix-engine developers

Reply:
My answer is yes. I'm not sure it is not thread-safe because I don't know the internal code, but when I tried to wrap this block with a mutex block, it worked without any problem.